synapse/rest/client/v2_alpha/account.py

```diff
@@ -884,7 +884,10 @@ def __init__(self, hs):
     async def on_GET(self, request):
         requester = await self.auth.get_user_by_req(request)
 
-        return 200, {"user_id": requester.user.to_string()}
+        return 200, {
+            "user_id": requester.user.to_string(),
+            "device_id": requester.device_id,
```

Comment:
Have you tested what it does if no device ID is associated with the access token (e.g. if the user belongs to an appservice)? It looks to me like it would include the `device_id` property in the response but with a `null` value, is this compliant with MSC2033? It might also be worth adding a test for that to make sure it doesn't randomly explode in the future.

Reply:
Do we have a test I can copy for being an appservice user?

Reply:
stole one from the appservice tests (which just makes sense, May 4th Travis)